zulipterminal/model.py

```diff
@@ -316,6 +318,22 @@ def get_topics_in_stream(self, stream_list: List[int]) -> None:
                 self.index['topics'][stream_id] = [topic['name'] for
                                                    topic in response['topics']]
 
+    def wrapper_get_topics(self, workers: int) -> None:
+        """
+        Distribute stream ids across threads in order to fetch
+        topics concurrently.
+        """
+        executor = ThreadPoolExecutor(max_workers=workers)
+        list_of_streams = list(self.stream_dict.keys())
+        thread_objects = []
+        for i in range(workers):
+            split_list = list_of_streams[i::workers]
+            thread_objects.append(executor.submit(
+                self.get_topics_in_stream, split_list))
+        # FIXME : Should we wait here for threads to
+        # complete?
+        # wait(thread_objects)
```

Comment:
In principle we may not need this: the code seems to work fine as it is; I've not reviewed enough yet to see whether this is since the UI spends time loading while this is happening (ie. we're lucky?) or that it really is unnecessary. I suppose it depends how quickly the UI might grab a topics list, relative to how fast the data is loading, and if that's an issue. Not waiting allows the code to carry on, which makes it appear faster.

If we use a contextmanager form of the ThreadPoolExecutor, then waiting happens automatically when it tidies up, but I think it would be OK to include an explicit wait if we want one. While we could add it later, this will also ensure that if we move to longer-lasting centralized threads for this kind of thing, then we still wait *here* (if intended).

Reply:
From what I observed, The topic fetch is pretty quick. However if we try to press `t` on a stream button as soon as the UI is loaded (1-2 seconds) no topics can be seen in the TopicView. In any case, I am now pushing a slight different version which handles thread safety also, albeit, its slightly slower (~6 seconds). This was partly the reason I didnt go with the context manager form initislly to shave around (2 seconds) of startup time.

Reply:
Startup takes ~8s for me (timing manually) with the current PR.
This compares to ~4-6s with master.

Note that the switch to the contextmanager form of the `ThreadPool` (as in the updated form now) means that it waits as part of the tidying up process; I tested switching that back and it seems to take 1-2s less time. The `wait` *seems* to make not much difference to this?

Reply:
Noted. Yes, I can repro the same timings as you. I read similarly, and was mostly sure the wait wasn't required (hence commented it). Will remove it in the upcoming iteration.